htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -416,3 +420,19 @@ def _inferDocumentDetailed(self, tokenList, sortResults=True):
 
     else:
       return categoryVotes, sampleIdList, distancetoSampleIds
+
+
+  @staticmethod
+  def _sortArray(array, seed=42):
+    """
+    Sort the input array, breaking ties in a random fashion.
+
+    @param array (numpy array)    Array of values to be sorted.
+    @param seed (int)             Seed the random number generator.
+
+    @return   (numpy array)       Sorted array indices, where the sort order is
+                                  greatest to least.
+    """
+    numpy.random.seed(seed)
+    randomValues = numpy.random.random(array.size)
+    return numpy.lexsort((randomValues, array))[::-1]
```

Comment:
I'm not familiar with lexsort - does it only use the randomValues for tie breaking?